pkg/stanza/fileconsumer/file.go

```diff
@@ -32,27 +32,40 @@ type EmitFunc func(ctx context.Context, attrs *FileAttributes, token []byte)
 
 type Manager struct {
 	*zap.SugaredLogger
-	wg     sync.WaitGroup
-	cancel context.CancelFunc
+	wg       sync.WaitGroup
+	workerWg sync.WaitGroup
+	cancel   context.CancelFunc
+	ctx      context.Context
 
 	readerFactory readerFactory
 	finder        Finder
 	roller        roller
 	persister     operator.Persister
 
 	pollInterval    time.Duration
-	maxBatches      int
```

Comment:
Well, I can raise another PR that applies a restriction, but on a number of files instead.
because we have no concept of batches in new approach

Reply:
We need to have an agreed upon design before a breaking change can be made. Then we can decide which order the changes are made.

Reply:
Sure,
what I currently have in mind is:
1). Revert back changes made by https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/3aeced40b0d5811af391385fd8696d36dafe59cc.
As we don't have any concept of batches in new approach.
2). I shall create a new PR which applies a similar limitation. We can limit number of files per poll cycle.

The reason I haven't combined that in this PR is that it'll create unnecessary confusion and both of the things are unrelated.

Reply:
we wouldn't put this check before consuming.
we will put it in the main loop, 
we will loop through all the paths, and send them to the channel, if and only if we're under the limit.
This way we can make sure that the file's are indeed consumed.

Reply:
Sorry I accidentally edited your comment instead of quoting it. I've restored it but may have lost some formatting.

---

> 2). I shall create a new PR which applies a similar limitation. We can limit number of files per poll cycle.

If you limit the number of files per poll cycle, how we would ensure that we are reading all the files?

Reply:
> we wouldn't put this check before consuming. we will put it in the main loop, we will loop through all the paths, and send them to the channel, if and only if we're under the limit. This way we can make sure that the file's are indeed consumed.

I don't think this addresses the problem I'm pointing out, but looking at `max_batches` again has me realizing that it doesn't either.

Let's say we have 100 files. If we set a limit of 20 files per poll cycle, then will we end up reading the same 20 files every poll cycle? If not, what is the mechanism for ensuring that we look at the other files?

Reply:
The glob returns the list of matched files in lexical order, I've verified it.
So yeah, we will keep consuming the same 20 files, unless some new ones are added or old ones are deleted.
The same problem is with https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/3aeced40b0d5811af391385fd8696d36dafe59cc

Reply:
We can try to introduce some randomness on the paths matched, this way we can avoid matching same files.
let me know what you think about this